### PR TITLE
Make `updateConfig` rebuild the engine only when new config is different from the old one

### LINKED ISSUE #1251: `updateConfig()` should check if new config is the same as the old one
### Description

`updateConfig()` is an expensive operation. It rebuilds the whole engine so we should avoid doing it when not necessary. This method should check if the new config is the same as the old one.

### Links

https://github.com/handsontable/handsontable/pull/10326
https://github.com/handsontable/hyperformula/issues/1252

### PR DESCRIPTION
**Pull request by sequba**
### Context

I added a check in `updateConfig` method. If the provided config changes nothing in the engine configuration, the engine is not rebuilt.

### How did you test your changes?

- unit tests passed
- new unit tests that verify the new feature

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
Fixes #1251

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [x] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: test/update-config.spec.ts [hunk 1]
```diff
@@ -67,4 +68,23 @@ describe('update config', () => {
     expect(engine.getCellValue(adr('A1'))).toBe(true)
     expect(engine.getCellValue(adr('B1'))).toBe(false)
   })
+
```
[the remark this answers is not in the document]
Comment (adamu-handsontable) [reply test/update-config.spec.ts:71]: I thought of another test that ensures that manual call of rebuildAndRecalculate works on an empty config as well:
```javascript
it('rebuilds the engine when manually called rebuildAndRecalculate on empty config', () => {
    const rebuildEngineSpy = spyOn(BuildEngineFactory, 'rebuildWithConfig')
    const engine = HyperFormula.buildFromArray([[]], {})

    engine.rebuildAndRecalculate()

    expect(rebuildEngineSpy).toHaveBeenCalled()
  })
```

**Review by adamu-handsontable — CHANGES_REQUESTED** [review]
I wrote 3 more unit tests that could be helpful to prevent potential problems when function logic is changed.

File: test/update-config.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sequba) [reply test/update-config.spec.ts:70]: Ok, I added these tests
